packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -555,21 +593,61 @@ class FlutterPlugin implements Plugin<Project> {
         }
     }
 
-    private Properties getPluginList() {
-        File pluginsFile = new File(project.projectDir.parentFile.parentFile, '.flutter-plugins')
-        Properties allPlugins = readPropertiesIfExist(pluginsFile)
-        Properties androidPlugins = new Properties()
-        allPlugins.each { name, path ->
-            if (doesSupportAndroidPlatform(path)) {
-                androidPlugins.setProperty(name, path)
+    /** Gets the list of plugins that support the Android platform. */
+    private List getPluginList() {
+        // Consider a `.flutter-plugins-dependencies` file with the following content:
+        // {
+        //     "plugins": {
+        //       "android": [
+        //         {
+        //           "name": "plugin-a",
+        //           "path": "/path/to/plugin-a",
+        //           "dependencies": ["plugin-b", "plugin-c"],
+        //           "native_build": true,
+        //         },
+        //         {
+        //           "name": "plugin-b",
+        //           "path": "/path/to/plugin-b",
+        //           "dependencies": ["plugin-c"],
+        //           "native_build": true,
+        //         },
+        //         {
+        //           "name": "plugin-c",
+        //           "path": "/path/to/plugin-c",
+        //           "dependencies": [],
+        //           "native_build": true,
+        //         },
+        //       ],
+        //     },
+        // }
+        // This means, `plugin-a` depends on `plugin-b` and `plugin-c`.
+        // `plugin-b` depends on `plugin-c`.
+        // `plugin-c` doesn't depend on anything.
+        Map meta = getDependenciesMetadata()
+        List androidPlugins = []
+        if (meta == null) {
+            return androidPlugins
+        }
+        assert meta.plugins instanceof Map
+        assert meta.plugins.android instanceof List
+
+        // This logic must be kept in sync with the logic in app_plugin_loader.groovy.
```

Comment:
Is it possible to share the code instead of just comment the logic must be the same?